### ISSUE TITLE
OpenShift Monitoring stuck at pending state - enabling errors with pod has unbound immediate PersistentVolumeClaims

### ISSUE THREAD

**Issue by eliassal** [crc-org/crc#3923]
### General information
  
   * OS: Windows Server 2022
   * Hypervisor: Hyper-V
   * Did you run crc setup before starting it (Yes/No)? Yes
   * Running CRC on: VM

## CRC version
```bash
# CRC version: 2.29.0+da5f55
```
  
## CRC status
```bash
level=debug msg="CRC version: 2.29.0+da5f55\n"
level=debug msg="OpenShift version: 4.14.1\n"
level=debug msg="Podman version: 4.4.4\n"
level=debug msg="Running 'crc status'"
CRC VM:          Running
OpenShift:       Starting (v4.14.1)
RAM Usage:       8.104GB of 12.5GB
Disk Usage:      26.74GB of 32.68GB (Inside the CRC VM)
Cache Usage:     25.11GB
Cache Directory: C:\Users\salam\.crc\cache
level=debug msg="Running '(Get-ItemProperty -Path \"HKLM:\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\" -Name ReleaseId).ReleaseId'"
level=debug msg="Running '(Get-ItemProperty -Path \"HKLM:\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\").EditionID'"

```

## CRC config
```bash
Modifies crc configuration properties.
Properties:

* bundle                                               Bundle path/URI - absolute or local path, http, https or docker URI (string, like 'https://foo.com/crc_hyperv_4.14.1_amd64.crcbundle', 'docker://quay.io/myorg/crc_hyperv_4.14.1_amd64.crcbundle:2.29.0' default 'C:\Users\salam\.crc\cache\crc_hyperv_4.14.1_amd64.crcbundle' )
* consent-telemetry                                    Consent to collection of anonymous usage data (yes/no)
* cpus                                                 Number of CPU cores (must be greater than or equal to '4')
* disable-update-check                                 Disable update check (true/false, default: false)
* disk-size                                            Total size in GiB of the disk (must be greater than or equal to '31')
* enable-bundle-quay-fallback                          If bundle download from the default location fails, fallback to quay.io (true/false, default: false)
* enable-cluster-monitoring                            Enable cluster monitoring Operator (true/false, default: false)
* enable-emergency-login                               Enable emergency login for 'core' user. Password is randomly generated. (true/false, default: false)
* enable-experimental-features                         Enable experimental features (true/false, default: false)
* enable-shared-dirs                                   Mounts host's user profile folder at '/' in the CRC VM (true/false, default: false)
* host-network-access                                  Allow TCP/IP connections from the CRC VM to services running on the host (true/false, default: false)
* http-proxy                                           HTTP proxy URL (string, like 'http://my-proxy.com:8443')
* https-proxy                                          HTTPS proxy URL (string, like 'https://my-proxy.com:8443')
* ingress-http-port                                    HTTP port to use for OpenShift ingress/routes on the host (1024-65535, default: 80)
* ingress-https-port                                   HTTPS port to use for OpenShift ingress/routes on the host (1024-65535, default: 443)
* kubeadmin-password                                   User defined kubeadmin password
* memory                                               Memory size in MiB (must be greater than or equal to '9216')
* nameserver                                           IPv4 address of nameserver (string, like '1.1.1.1 or 8.8.8.8')
* no-proxy                                             Hosts, ipv4 addresses or CIDR which do not use a proxy (string, comma-separated list such as '127.0.0.1,192.168.100.1/24')
* persistent-volume-size                               Total size in GiB of the persistent volume used by the CSI driver for microshift preset (must be greater than or equal to '15')
* preset                                               Virtual machine preset (valid values are: [podman openshift okd microshift])
* proxy-ca-file                                        Path to an HTTPS proxy certificate authority (CA)
* pull-secret-file                                     Path of image pull secret (download from https://console.redhat.com/openshift/create/local)
* shared-dir-password                                  Password used while using CIFS/SMB file sharing (It is the password for the current logged in user)
* skip-check-admin-helper-service-running              Skip preflight check (true/false, default: false)
* skip-check-administrator-user                        Skip preflight check (true/false, default: false)
* skip-check-bundle-extracted                          Skip preflight check (true/false, default: false)
* skip-check-crc-users-group-exists                    Skip preflight check (true/false, default: false)
* skip-check-daemon-task-install                       Skip preflight check (true/false, default: false)
* skip-check-daemon-task-posh-script-present           Skip preflight check (true/false, default: false)
* skip-check-daemon-task-running                       Skip preflight check (true/false, default: false)
* skip-check-hyperv-installed                          Skip preflight check (true/false, default: false)
* skip-check-hyperv-service-running                    Skip preflight check (true/false, default: false)
* skip-check-ram                                       Skip preflight check (true/false, default: false)
* skip-check-user-in-crc-users-and-hyperv-admins-group Skip preflight check (true/false, default: false)
* skip-check-vsock                                     Skip preflight check (true/false, default: false)
* skip-check-windows-edition                           Skip preflight check (true/false, default: false)
* skip-check-windows-version                           Skip preflight check (true/false, default: false)

Usage:
  crc config SUBCOMMAND [flags]
  crc config [command]

Available Commands:
  get         Get a crc configuration property
  set         Set a crc configuration property
  unset       Unset a crc configuration property
  view        Display all assigned crc configuration properties

```

## Host Operating System
```bash
Host Name:                 SALAMCOG
OS Name:                   Microsoft Windows Server 2019 Datacenter
OS Version:                10.0.17763 N/A Build 17763
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Member Server
OS Build Type:             Multiprocessor Free
Registered Owner:          Windows User
Registered Organization:
Product ID:                00430-70395-99300-AA857
Original Install Date:     01/05/2020, 19:03:36
System Boot Time:          23/11/2023, 10:53:10
System Manufacturer:       Supermicro
System Model:              Super Server
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 60 Stepping 3 GenuineIntel ~3401 Mhz
BIOS Version:              American Megatrends Inc. 3.2, 25/05/2018
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume5
System Locale:             fr;French (France)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC+01:00) Brussels, Copenhagen, Madrid, Paris
Total Physical Memory:     32 651 MB
Available Physical Memory: 9 866 MB
Virtual Memory: Max Size:  72 651 MB
Virtual Memory: Available: 49 737 MB
Virtual Memory: In Use:    22 914 MB
Page File Location(s):     C:\pagefile.sys
                           H:\pagefile.sys
Domain:                    salam.net
Logon Server:              \\SALAMMAIN
Hotfix(s):                 38 Hotfix(s) Installed.
                           [01]: KB5031984
                           [02]: KB4535680
                           [03]: KB4549947
                           [04]: KB4558997
                           [05]: KB4562562
                           [06]: KB4566424
                           [07]: KB4570332
                           [08]: KB4577586
                           [09]: KB4577667
                           [10]: KB4587735
                           [11]: KB4589208
                           [12]: KB4598480
                           [13]: KB4601393
                           [14]: KB5000859
                           [15]: KB5001404
                           [16]: KB5001407
                           [17]: KB5003243
                           [18]: KB5003711
                           [19]: KB5004424
                           [20]: KB5005112
                           [21]: KB5032196
                           [22]: KB5006754
                           [23]: KB5008539
                           [24]: KB5008287
                           [25]: KB5009642
                           [26]: KB5011574
                           [27]: KB5012675
                           [28]: KB5014031
                           [29]: KB5014797
                           [30]: KB5015896
                           [31]: KB5017400
                           [32]: KB5020374
                           [33]: KB5023789
                           [34]: KB5028316
                           [35]: KB5030505
                           [36]: KB5031589
                           [37]: KB5032306
                           [38]: KB5005701
Network Card(s):           8 NIC(s) Installed.
                           [01]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (vExt-Eth1)
                                 Status:          Hardware not present
                           [02]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (vExternal-Eth2)
                                 Status:          Hardware not present
                           [03]: Intel(R) Ethernet Connection I217-LM
                                 Connection Name: Eth1-up
                                 DHCP Enabled:    Yes
                                 DHCP Server:     N/A
                                 IP address(es)
                           [04]: Intel(R) I210 Gigabit Network Connection
                                 Connection Name: Ethernet 2
                                 Status:          Media disconnected
                           [05]: Realtek PCI GbE Family Controller
                                 Connection Name: Eth1
                                 DHCP Enabled:    Yes
                                 DHCP Server:     N/A
                                 IP address(es)
                           [06]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (OnEth1-Up)
                                 DHCP Enabled:    Yes
                                 DHCP Server:     192.168.1.1
                                 IP address(es)
                                 [01]: 192.168.1.10
                           [07]: Realtek USB GbE Family Controller
                                 Connection Name: Ethernet 3
                                 DHCP Enabled:    Yes
                                 DHCP Server:     N/A
                                 IP address(es)
                           [08]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (USB-VS)
                                 DHCP Enabled:    Yes
                                 DHCP Server:     192.168.1.1
                                 IP address(es)
                                 [01]: 192.168.1.14
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.

```

### Steps to reproduce

  1. Enable openshift monitoring by 
  2. oc patch clusterversion/version --type='json' -p '[{"op":"remove", "path":"/spec/overrides/0"}]' -oyaml
  3. Issuing oc get pods I see
 NAME                                                     READY   STATUS    RESTARTS   AGE
alertmanager-main-0                                      6/6     Running   12         24h
cluster-monitoring-operator-9545fdbb-gkd2t               1/1     Running   2          24h
kube-state-metrics-76c49598db-lb5p8                      3/3     Running   6          24h
monitoring-plugin-65d46449bb-gk5lm                       1/1     Running   2          24h
node-exporter-qtgnd                                      2/2     Running   4          24h
openshift-state-metrics-587f959b46-44fs2                 3/3     Running   6          24h
prometheus-adapter-7994675c6b-vdn4t                      1/1     Running   2          24h
**prometheus-k8s-0                                         0/6     Pending   0          24h**
prometheus-operator-59c99bbf7c-m9x72                     2/2     Running   4          24h
prometheus-operator-admission-webhook-587674b4bc-7sgt4   1/1     Running   2          24h
telemeter-client-b7ff685c9-6jcvf                         3/3     Running   6          23h
thanos-querier-64bf78f6ff-9792m                          6/6     Running   12         24h
  5. prometheus-k8s-0  pods to be in ready 6/6 which is not the case
  6. I do oc describe pod prometheus-k8s-0 at the end of the output I see
  Events:
  Type     Reason            Age                   From               Message
  ----     ------            ----                  ----               -------
  Warning  FailedScheduling  14h (x481 over 22h)   default-scheduler  0/1 nodes are available: **pod has unbound immediate PersistentVolumeClaims. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling..**
  Warning  FailedScheduling  4m54s (x59 over 60m)  default-scheduler  0/1 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling..

### Expected
prometheus-k8s-0  pods to be in ready 6/6
### Actual
prometheus-k8s-0  pods not going through 0/6

**Comment by gbraad** [comment]
@praveenkumar @jsliacan Seen this before?

**Comment by gbraad** [comment]
is this consistently reproducible?

what is set for `crc config view`?

**Comment by eliassal** [comment]
Yes, even IO started this morning the cluster with 15 G of memopry and 50 g of space drive
`crc start -d 50 -m 15288 -c 4`

Always same thing
here is the ouput of 

> crc config view
> - consent-telemetry                     : yes
> - enable-cluster-monitoring             : true
>

**Comment by anjannath** [comment]
@eliassal i tried to reproduce but for me all the pods were running, if the issue is with PersistentVolumes, can you please try to create a simple pod with a pvc to check if it is a general issue with storage or something with the monitoring operator

```
PS C:\Users\anath> oc get pods -n openshift-monitoring
NAME                                                     READY   STATUS    RESTARTS   AGE
alertmanager-main-0                                      6/6     Running   0          52m
cluster-monitoring-operator-9545fdbb-r6hfm               1/1     Running   0          53m
kube-state-metrics-76c49598db-7r9wg                      3/3     Running   0          52m
monitoring-plugin-65d46449bb-86bjp                       1/1     Running   0          52m
node-exporter-jtm5t                                      2/2     Running   0          52m
openshift-state-metrics-587f959b46-stgc6                 3/3     Running   0          52m
prometheus-adapter-5c4d7844d4-sc5hq                      1/1     Running   0          52m
prometheus-k8s-0                                         6/6     Running   0          52m
prometheus-operator-59c99bbf7c-8xbd9                     2/2     Running   0          52m
prometheus-operator-admission-webhook-587674b4bc-m92sx   1/1     Running   0          52m
telemeter-client-5445dcdb6b-fchh4                        3/3     Running   0          52m
thanos-querier-66b955b68c-s9j4q                          6/6     Running   0          52m
```

The events for the `prometheus-k8s-0` also doesn't have the error like you mentioned.
```
PS C:\Users\anath> oc events --for pod/prometheus-k8s-0 -n openshift-monitoring
LAST SEEN   TYPE     REASON           OBJECT                 MESSAGE
37m         Normal   Scheduled        Pod/prometheus-k8s-0   Successfully assigned openshift-monitoring/prometheus-k8s-0 to crc-n5gv4-master-0
37m         Normal   AddedInterface   Pod/prometheus-k8s-0   Add eth0 [10.217.0.72/23] from openshift-sdn
37m         Normal   Pulling          Pod/prometheus-k8s-0   Pulling image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:3f89e9943526348b2191826004851cd2c46aabd5ffb683890a23a193610e2449"
37m         Normal   Pulled           Pod/prometheus-k8s-0   Successfully pulled image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:3f89e9943526348b2191826004851cd2c46aabd5ffb683890a23a193610e2449" in 6.11918162s (6.11919372s including waiting)
37m         Normal   Created          Pod/prometheus-k8s-0   Created container init-config-reloader
37m         Normal   Started          Pod/prometheus-k8s-0   Started container init-config-reloader
37m         Normal   Pulling          Pod/prometheus-k8s-0   Pulling image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:a3b8b44ae7a81671b36432ce7b0c0c82bf801d69d8e8a8b69d76d02d786e517f"
37m         Normal   Pulled           Pod/prometheus-k8s-0   Successfully pulled image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:a3b8b44ae7a81671b36432ce7b0c0c82bf801d69d8e8a8b69d76d02d786e517f" in 11.888822184s (11.888844384s including waiting)
37m         Normal   Created          Pod/prometheus-k8s-0   Created container prometheus
37m         Normal   Started          Pod/prometheus-k8s-0   Started container prometheus
37m         Normal   Pulled           Pod/prometheus-k8s-0   Container image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:3f89e9943526348b2191826004851cd2c46aabd5ffb683890a23a193610e2449" already present on machine
37m         Normal   Created          Pod/prometheus-k8s-0   Created container config-reloader
37m         Normal   Started          Pod/prometheus-k8s-0   Started container config-reloader
37m         Normal   Pulled           Pod/prometheus-k8s-0   Container image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:22886999eb60062334a345a2fd53054a14fe6e3ad3788d83ce5287742561cc03" already present on machine
37m         Normal   Created          Pod/prometheus-k8s-0   Created container thanos-sidecar
37m         Normal   Started          Pod/prometheus-k8s-0   Started container thanos-sidecar
37m         Normal   Pulled           Pod/prometheus-k8s-0   Container image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:912a7ebfdff429faf7be4990cce9dd149eff8b9d1635289f945be69c2b91e7c2" already present on machine
37m         Normal   Created          Pod/prometheus-k8s-0   Created container prometheus-proxy
37m         Normal   Started          Pod/prometheus-k8s-0   Started container prometheus-proxy
37m         Normal   Pulled           Pod/prometheus-k8s-0   Container image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:8db8a3708497fb6a8ef42faaa1bcc131af33347bc5960d368d401b77d5d9ba74" already present on machine
37m         Normal   Created          Pod/prometheus-k8s-0   Created container kube-rbac-proxy
37m         Normal   Started          Pod/prometheus-k8s-0   Started container kube-rbac-proxy
37m         Normal   Pulled           Pod/prometheus-k8s-0   Container image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:8db8a3708497fb6a8ef42faaa1bcc131af33347bc5960d368d401b77d5d9ba74" already present on machine
37m         Normal   Created          Pod/prometheus-k8s-0   Created container kube-rbac-proxy-thanos
37m         Normal   Started          Pod/prometheus-k8s-0   Started container kube-rbac-proxy-thanos
```

I had enabled the monitoring stack using `crc config set enable-cluster-monitoring true`
```
PS C:\Users\anath> crc config view
- consent-telemetry                     : no
- cpus                                  : 5
- enable-cluster-monitoring             : true
- memory                                : 16000
```

**Comment by eliassal** [comment]
Hi @anjannath, I visited the storage blade, there are 4 items, 3 of them are in bound status (created after I enabled prometheus) the 4th one which is prometheus-k8s-0 db-prometheus-k8s0. Here is a snapshot

![image](https://github.com/crc-org/crc/assets/8447864/e375963f-d92a-4d01-afa3-ce277c00a0ff)

**Comment by eliassal** [comment]
Here is the output of  crc config view
```
- consent-telemetry                     : yes
- enable-cluster-monitoring             : true
```

**Comment by eliassal** [comment]
And here ios the output for 
 `oc` events --for pod/prometheus-k8s-0 -n openshift-monitoring

```
LAST SEEN                TYPE      REASON             OBJECT                 MESSAGE
2d21h (x722 over 3d9h)   Warning   FailedScheduling   Pod/prometheus-k8s-0   0/1 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling..
19m (x2 over 21m)        Warning   FailedScheduling   Pod/prometheus-k8s-0   0/1 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling..

```

**Comment by anjannath** [comment]
@eliassal I checked the `pvcs` and i don't have any after enabling monitoring, looking at the screenshot you shared the pending pvc is using the storage class `fast` which is not present in the crc cluster, it should use `crc-csi-hostpath-provisioner` SC like the other bound pvcs. See screenshot below:

![image](https://github.com/crc-org/crc/assets/8885742/0a280da3-edf9-47d6-a3ce-d6b312cd11c7)


since i am not able to reproduce the issue you are facing, can you start from scratch, as you also mentioned you ran `oc patch clusterversion/version --type='json' -p '[{"op":"remove", "path":"/spec/overrides/0"}]' -oyaml` to enable monitoring, which should not be needed, enabling it from crc config should be enough

To start from a clean env delete the existing crc instance using `crc delete -f` and then start with `crc start -p <pull_secret>`

Then in the webconsole check if you have a new Tab on the sidebar named `Observe` like the below screenshot:
![image](https://github.com/crc-org/crc/assets/8885742/22749917-2618-4565-b9b0-4c0982744693)

**Comment by eliassal** [comment]
OK I will do but as a first hint, when I started the crc to start playing with it, I di not have the "Observe" in the menu that is why I followed a video. I will let you know

**Comment by eliassal** [comment]
@anjannath Yeees, I did
 ```
crc delete -f
 crc setup
 crc start -d 50 -m 15288 -c 4 -p ....../crc-windows-secret.txt
 crc config set enable-cluster-monitoring true
```

Browsed to console, and OBSERVE was there and I had metrics in the dashboard. But tell me please for curiosity, so what is/means/when it should be used 
`oc patch clusterversion/version --type='json' -p '[{"op":"remove", "path":"/spec/overrides/0"}]' -oyaml`
that I found in one of the tutorials, and why It used the class `fast` instead of `crc-csi-hostpath-provisioner.`
![image](https://github.com/crc-org/crc/assets/8447864/3542f665-4873-41c8-ac3b-8ceab8bb73ae)

**Comment by anjannath** [comment]
> Browsed to console, and OBSERVE was there and I had metrics in the dashboard. But tell me please for curiosity, so what is/means/when it should be used
oc patch clusterversion/version --type='json' -p '[{"op":"remove", "path":"/spec/overrides/0"}]' -oyaml
that I found in one of the tutorials, and why It used the class fast instead of crc-csi-hostpath-provisioner.

not sure what tutorial you are referring to, maybe it has outdated information, our documentation doesn't mention this, see: https://crc.dev/crc/getting_started/getting_started/administrative-tasks/#_starting_monitoring

for the storage class it was a general observation that it was trying to use a non-existing one, maybe the tutorial you followed had some other setup.

Closing as the original issue is resolved.
